src/hd/hd.c

```diff
@@ -6067,16 +6067,26 @@ char *get_sysfs_attr_by_path(const char *path, const char *attr)
  */
 char *get_sysfs_attr_by_path2(const char *path, const char *attr, unsigned *len)
 {
-  static char buf[1024];
-  int i, fd;
+  static char *buf = NULL;
+  char * ptr;
+  int i, fd, max;
 
   if(len) *len = 0;
 
+  // init static buffer on first run
+  if(!buf) buf = new_mem(MAX_ATTR_SIZE + 1);
```

Comment:
OK, but what if `libhd` is used from another application like YaST? Would that make sense?